src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -829,6 +830,102 @@ ov::pass::NopSliceBeforeGatherElements::NopSliceBeforeGatherElements() {
     register_matcher(m, matcher_pass_callback);
 }
 
+ov::pass::NopStridedSlice::NopStridedSlice() {
+    MATCHER_SCOPE(NopStridedSlice);
+
+    auto input = pattern::any_input();
+    auto begin_const = pattern::wrap_type<ov::op::v0::Constant>();
+    auto end_const = pattern::wrap_type<ov::op::v0::Constant>();
+    auto optional_stride_const = pattern::optional<ov::op::v0::Constant>();
+    auto pattern = pattern::wrap_type<ov::op::v1::StridedSlice>({input, begin_const, end_const, optional_stride_const});
+
+    ov::matcher_pass_callback matcher_pass_callback = [=](pattern::Matcher& m) {
+        auto node = m.get_match_root();
+        if (node == nullptr) {
+            return false;
+        }
+        auto strided_slice_node = std::dynamic_pointer_cast<ov::op::v1::StridedSlice>(node);
+        if (!strided_slice_node->get_shrink_axis_mask().empty() || !strided_slice_node->get_new_axis_mask().empty() ||
+            !strided_slice_node->get_ellipsis_mask().empty()) {
+            return false;
+        }
+        if (!op::util::is_constant_and_all_values_equal_int(node->input_value(3), 1) &&
+            node->get_input_partial_shape(3).size() != 0) {
+            return false;
+        }
+        auto begin_node = strided_slice_node->get_input_node_shared_ptr(1);
+        if (const auto& begin_constant_node = ov::util::get_constant_from_source(begin_node)) {
+            const auto& values = begin_constant_node->cast_vector<int64_t>();
+            auto begin_mask = strided_slice_node->get_begin_mask();
+            const auto dim_size = begin_mask.size();
+            if (values.size() != dim_size) {
+                return false;
+            }
+            for (size_t i = 0; i < dim_size; ++i) {
+                if (!begin_mask[i] && values[i]) {
+                    return false;
+                }
+            }
+        }
+
+        auto end_node = strided_slice_node->get_input_node_shared_ptr(2);
+        if (const auto& end_constant_node = ov::util::get_constant_from_source(end_node)) {
```

Comment:
should we return false if it's not constant?

Reply:
fixed